### DISCUSSION TITLE
More "avanced" annoncer message templates

### DISCUSSION THREAD

**Discussion by cyrillesondag** [jreleaser/jreleaser#1079] [Ideas]
hi,

In annoncers (at least in Mattermost ) wouldn't it be more richfull to templatize the raw post body ?

So like this you could have  either a "simple text" message with "text" property, or a more elaborate message with the "template"

For exemple some thing like this :
```
Object message;
if (isNotBlank(mattermost.getMessage())) {
    message = Message.of(mattermost.getResolvedMessage(context));
} else {
    Map<String, Object> props = new LinkedHashMap<>();
    props.put(Constants.KEY_CHANGELOG, MustacheUtils.passThrough(context.getChangelog()));
    context.getModel().getRelease().getReleaser().fillProps(props, context.getModel());
    message = mattermost.getResolvedMessageTemplate(context, props);  // raw json message body
}
```

**Comment by aalmiray** [comment]
Aha I see what you mean now. This has a relationship with https://github.com/jreleaser/jreleaser/issues/945 as well.

The proposed change would break behavior compatibility unless it's guarded by a configuration flag `rawMessageTemplate` (better name is needed) so when active it would offer the behavior you seek.

**Reply by cyrillesondag** [reply]
yes indeed, but only for those who use templates who will have to provide a json formatted template.

IMO one "simple" and one "advanced" solution would be enought (for the good of the api), But yes.. i get your point.

**Reply by aalmiray** [reply]
Filed https://github.com/jreleaser/jreleaser/issues/1080

**Reply by aalmiray** [reply]
And fixed.

**Reply by cyrillesondag** [reply]
wow ! thanks
